src/webportal/src/app/job/job-view/fabric/job-detail.jsx

```diff
@@ -273,6 +285,53 @@ class JobDetail extends React.Component {
     }
   }
 
+  generateTransferState(tags) {
+    try {
+      // find out successfully transferred beds
+      const transferredPrefix = 'pai-transferred-to-';
+      const transferredURLs = [];
+      const transferredClusterSet = new Set();
+      for (let tag of tags) {
+        if (tag.startsWith(transferredPrefix)) {
+          tag = tag.substr(transferredPrefix.length);
+          const urlPosition = tag.lastIndexOf('-url-');
+          if (urlPosition !== -1) {
+            transferredClusterSet.add(tag.substr(0, urlPosition));
+            transferredURLs.push(tag.substr(urlPosition + 5));
+          }
+        }
+      }
+      // find out failed transfer attempts
+      const transferAttemptPrefix = 'pai-transfer-attempt-to-';
```

Comment:
Will a job has two tags ```pai-transferred-to-```, ```pai-transfer-attempt-to-```  in the meanwhile?

Reply:
Yes. If a job is transferred successfully, it will have both `pai-transferred-to-` and `pai-transfer-attempt-to-`. If there is error during transfer, it will only have `pai-transfer-attempt-to-`, and no  `pai-transferred-to-`. Both cases will be handled properly.